src/irmin-pack/inode.mli

```diff
@@ -23,31 +23,31 @@ end
 module type S = sig
   include Irmin.CONTENT_ADDRESSABLE_STORE
 
+  type index
+
   val v :
     ?fresh:bool ->
     ?shared:bool ->
     ?readonly:bool ->
     ?lru_size:int ->
+    index:index ->
     string ->
     [ `Read ] t Lwt.t
 
   val batch : [ `Read ] t -> ([ `Read | `Write ] t -> 'a Lwt.t) -> 'a Lwt.t
 
   module Key : Irmin.Hash.S with type t = key
 
-  module Val :
-    Irmin.Private.Node.S
-    with type t = value
-     and type hash = key
-     and type t = value
+  module Val : Irmin.Private.Node.S with type t = value and type hash = key
 end
 
 module Make
     (Conf : CONFIG)
-    (P : Pack.MAKER)
-    (H : Irmin.Hash.S with type t = P.key)
+    (H : Irmin.Hash.S)
+    (P : Pack.MAKER with type key = H.t and type index = Pack_index.Make(H).t)
```

Comment:
same here, `type index = Pack_index.Make(H).t` might be unecessary

Reply:
This one is necessary, I believe. Removing it causes failure to satisfy [the return type constraint](https://github.com/mirage/irmin/pull/805/files/343a4d7d75a6e26a801de601895d012ee480db68#diff-38713f52e17cb80446685ae883d6b202R53), which is in turn necessary to unify `Index.t` with (e.g.) `Node.CA.index`.